From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct us_data
Patch-mainline: never

In 3.0.23, commit 721eaa34e5 (usb-storage: fix freezing of the
scanning thread), upstream commit bb94a40668277, removed scanning_done
completion and added scan_dwork delayed_work instead. It indeed
changed layout of that structure.

Let's just ignore it because this is an internal USB storage structure.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/usb/storage/usb.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/storage/usb.h
+++ b/drivers/usb/storage/usb.h
@@ -149,7 +149,11 @@ struct us_data {
 	struct completion	cmnd_ready;	 /* to sleep thread on	    */
 	struct completion	notify;		 /* thread begin/end	    */
 	wait_queue_head_t	delay_wait;	 /* wait during reset	    */
+#ifdef __GENKSYMS__
+	struct completion	scanning_done;
+#else
 	struct delayed_work	scan_dwork;	 /* for async scanning      */
+#endif
 
 	/* subdriver information */
 	void			*extra;		 /* Any extra data          */
